var/spack/repos/builtin/packages/py-netcdf4/package.py

```diff
@@ -23,3 +25,17 @@ class PyNetcdf4(PythonPackage):
 
     depends_on('netcdf')
     depends_on('hdf5@1.8.0:')
+
+    def setup_environment(self, spack_env, run_env):
+        """Ensure installed netcdf and hdf5 libraries are used"""
+        hdf5_include_dir = os.path.dirname(self.spec['hdf5'].headers[0])
+        hdf5_lib_dir = os.path.dirname(self.spec['hdf5'].libs[0])
+        netcdf_include_dir = os.path.dirname(self.spec['netcdf'].headers[0])
+        netcdf_lib_dir = os.path.dirname(self.spec['netcdf'].libs[0])
```

Comment:
I believe there are more Spack-idoimatic ways to get these.  Try:
```
        spack_env.set('USE_SETUPCFG', '0')
        spack_env.set('HDF5_INCDIR',self.spec['hdf5'].prefix.include)
        spack_env.set('HDF5_LIBDIR', self.spec['hdf5'].prefix.lib)
        ...etc for netcdf too....
```

Reply:
This makes sense, I was trying to figure out how to do this from the documentation, but looking through other packages would have helped.